src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultActions = [];
+  let newState = { ...state };
+
+  for (const action of actions) {
+    const { type } = action;
+    const obj = { ...newState };
+
+    if (type === 'addProperties') {
+      const { extraData } = action;
+
+      Object.assign(obj, extraData);
+    }
+
+    if (type === 'removeProperties') {
+      const { keysToRemove } = action;
+
+      for (const key of keysToRemove) {
+        delete obj[key];
+      }
+    }
+
+    if (type === 'clear') {
+      for (const key in obj) {
+        delete obj[key];
+      }
+    }
+
+    newState = obj;
```

Comment:
you can just use `Object.assign` and add a new property to `newState`
instead of creating a new object and then reassign `newState`